src/lib/y2packager/solvable_pool.rb

```diff
@@ -27,12 +27,11 @@ def initialize
     # @param name [String] Name of the repository
     def add_rpmmd_repo(primary_xml, name)
       repo = pool.add_repo(name)
-      gz = open(primary_xml)
-      fd = Solv.xfopen_fd(primary_xml, gz.fileno)
-      repo.add_rpmmd(fd, nil, 0)
-      pool.createwhatprovides
-    ensure
-      gz.close if gz
+      File.open(primary_xml) do |gz|
+        fd = Solv.xfopen_fd(primary_xml, gz.fileno)
+        repo.add_rpmmd(fd, nil, 0)
+        pool.createwhatprovides
```

Comment:
NP: I'd just move this line outside the block, it's not related to the opened file.

I completely forgot that I could use the block here...